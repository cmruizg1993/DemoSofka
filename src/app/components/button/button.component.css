.btn {
    border-radius: var(--size-xs);
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    font-family: "DemoFont";
    font-size: var(--size-sm);
    cursor: pointer;
    width: max-content;
    border: 0;
    padding: var(--size-sm);
    background-color: var(--darkGrayishBlue300);
}
.btn:hover{
    background-color: var(--darkGrayishBlue500);
    color: var(--blue600);
    border: 0.01px solid rgba(red, green, blue, alpha);
    font-weight: bold;
}

.btn-primary {
    background-color: var(--yellow600) !important;
}
.btn-primary:hover{
    background-color: var(--yellow800) !important;
    color: var(--blue600);
    border: 0.01px solid rgba(red, green, blue, alpha);
    font-weight: bold;
}
.btn-primary:disabled,
.btn-primary[disabled]{
    background-color: var(--yellow200) !important;
    color: var(--blue400);
    border: 0.01px solid rgba(red, green, blue, alpha);
    font-weight: initial !important;
}

