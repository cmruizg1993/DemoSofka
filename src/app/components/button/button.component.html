<button type="button" class="btn" 
[class]="class" 
[class.btn-primary]="primary" 
[disabled]="disabled"
(click)="buttonClick()">{{label}}</button>
