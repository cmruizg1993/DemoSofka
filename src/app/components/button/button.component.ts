import { Component, EventEmitter, Input, Output } from '@angular/core';

@Component({
  selector: 'app-button',
  templateUrl: './button.component.html',
  styleUrl: './button.component.css'
})
export class ButtonComponent {
  @Input() label = 'Bot√≥n'
  @Input() primary?: boolean;
  @Input() disabled?: boolean ;
  @Input() class = '';
  @Output() onClick = new EventEmitter();

  buttonClick(){
    this.onClick.emit();
  }
}
