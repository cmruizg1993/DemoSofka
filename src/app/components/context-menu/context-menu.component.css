.context-menu {
    background-color: #fafafa;
    padding: 4pt;
    font-size: 10pt;
    z-index: 1000;
    box-shadow: 0 0 12pt rgba(0, 0, 0, 0.25);
    border-radius: 4pt;
    padding: 0.5em 0 0.5em 0;
    animation: fadeIn 0.1s ease-out;
    opacity:1.0;
    display:block;
    position: absolute;
  }
  
  
  .context-menu hr {
    border: none;
    border-bottom: 1px solid #eee;
  }
  
  .context-menu div {
    cursor: pointer;
    display: block;
    text-decoration: none;
    color: #333;
    padding: 0.5em 2em 0.5em 0.75em;
    max-width: 18em;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }
  
  .context-menu div:hover {
    background-color: var(--blue800);
    color: #fff;
  }
  
  
  .context-menu div::before {
    content: '';
    margin-right: 0.75em;
    width: 0.5em;
    height: 1em;
    display: inline-block;
  }