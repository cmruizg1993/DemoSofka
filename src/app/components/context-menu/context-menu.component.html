<ng-template #context let-context>
	<section class="context-menu">
		<div *ngFor="let item of contextMenu.items" (click)="executeCallBack(item)">{{item.label}}</div>
	</section>
</ng-template>