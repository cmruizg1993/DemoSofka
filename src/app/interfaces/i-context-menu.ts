export interface IContextMenu {
    data?: any;
    items: IContextMenuItem[];
}

export interface IContextMenuItem {
    label: string;
    action: CallableFunction;
}
