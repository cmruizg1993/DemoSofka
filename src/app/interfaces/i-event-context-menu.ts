export interface IEventContextMenu {
    event: MouseEvent;
    data: any;
}
