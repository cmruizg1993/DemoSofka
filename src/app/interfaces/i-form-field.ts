import { AsyncValidatorFn, ValidatorFn } from "@angular/forms";

export interface IFormField {
    name: string;
    placeholder?:string;
    label?: string;
    required?: boolean;
    pattern?: string;
    female?: boolean;
    onValidation?: CallableFunction;
    asynValidators?: AsyncValidatorFn[];
    validators?: ValidatorFn[];
    disabled?: boolean;
    value?: any;
}
