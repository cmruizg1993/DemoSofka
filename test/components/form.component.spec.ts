import { ComponentFixture, TestBed } from '@angular/core/testing';
import { ContextMenuComponent } from '../../src/app/components/context-menu/context-menu.component';
import { RouterTestingModule } from '@angular/router/testing';
import { IContextMenu } from '../../src/app/interfaces/i-context-menu';
import { Observable, Subject } from 'rxjs';
import { IEventContextMenu } from '../../src/app/interfaces/i-event-context-menu';
import { By } from '@angular/platform-browser';
import { FormComponent } from '../../src/app/components/form/form.component';
import { ButtonComponent } from '../../src/app/components/button/button.component';
import { IFormField } from '../../src/app/interfaces/i-form-field';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';



describe('FormComponent', () => {
  let fixture: ComponentFixture<FormComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [
        RouterTestingModule,
        ReactiveFormsModule,
        FormsModule
      ],
      declarations: [
        FormComponent,
        ButtonComponent
      ],
    }).compileComponents();
    fixture = TestBed.createComponent(FormComponent);
    const fields: IFormField[] = [
      {
        name: 'Test', pattern: '[0-9]{1,1}', value: 'test'
      }
    ]
    fixture.componentInstance.fields = fields;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  it('Se debe crear el componente', () => {
    const component = fixture.componentInstance;
    expect(component).toBeTruthy();
  });
  it(`Debe coincidir con el snapshot'`, () => {
    expect(compiled).toMatchSnapshot();
  });
  it(`Debe resetear el formulario'`, () => {
    const component = fixture.componentInstance;
    
    component.buttons = [
      {
        label: 'test', reset: true
      }
    ]
    component.onClick(component.buttons[0]);

    expect(true).toBeTruthy();
  });

});
