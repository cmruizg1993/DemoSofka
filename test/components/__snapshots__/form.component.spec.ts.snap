// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormComponent Debe coincidir con el snapshot' 1`] = `
<div
  id="root1"
>
  <form
    class="ng-untouched ng-pristine ng-invalid"
    novalidate=""
  >
    <div
      class="row"
    >
      <div
        class="col-md-6"
      >
        <div
          class="form-group p-2"
        >
          <label
            class="text-primary fs-2 fw-800"
          >
            
          </label>
          <input
            class="w100 ng-untouched ng-pristine ng-invalid"
            placeholder=""
            type="text"
          />
          <div
            class="text-danger fw-bolder"
            style="height: 1rem !important;"
          >
            
          </div>
        </div>
      </div>
      
    </div>
    <div
      class="row p-4"
    >
      
    </div>
  </form>
</div>
`;
